provider/activitypub/mastodon/client.go

```diff
@@ -1,67 +1,299 @@
 package mastodon
 
```

Comment:
There are no visible tests for these methods. Consider adding unit tests to ensure correct behavior.